cms/djangoapps/contentstore/views/course_optimizer.py

```diff
@@ -0,0 +1,218 @@
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_GET, require_http_methods
+from opaque_keys.edx.keys import CourseKey
+from user_tasks.conf import settings as user_tasks_settings
+from user_tasks.models import UserTaskArtifact, UserTaskStatus
+
+from common.djangoapps.student.auth import has_course_author_access
+from common.djangoapps.util.json_request import JsonResponse
+from common.djangoapps.util.views import ensure_valid_course_key
+from cms.djangoapps.contentstore.xblock_storage_handlers.view_handlers import get_xblock
+from cms.djangoapps.contentstore.xblock_storage_handlers.xblock_helpers import usage_key_with_run
+from xmodule.modulestore.django import modulestore  # lint-amnesty, pylint: disable=wrong-import-order
+
+from ..tasks import CourseLinkCheckTask, check_broken_links
+from ..utils import reverse_course_url
+
+__all__ = [
+    'link_check_handler',
+    'link_check_status_handler',
+]
+
+log = logging.getLogger(__name__)
+
+STATUS_FILTERS = user_tasks_settings.USER_TASKS_STATUS_FILTERS
+
+
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@login_required
+@require_http_methods(('GET', 'POST'))
+@ensure_valid_course_key
+def link_check_handler(request, course_key_string):
+    """
+    The restful handler for checking broken links in a course.
+
+    GET
+        html: return html page for import page ???
+        json: not supported ???
+    POST
+        Start a Celery task to check broken links in the course
+
+    The Studio UI uses a POST request to start the link check asynchronously, with
+    a link appearing on the page once it's ready.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not has_course_author_access(request.user, course_key):
+        raise PermissionDenied()
+    courselike_block = modulestore().get_course(course_key)
+    if courselike_block is None:
+        raise Http404
+    context = {
+        'context_course': courselike_block,
+        'courselike_home_url': reverse_course_url("course_handler", course_key),
+    }
+    context['status_url'] = reverse_course_url('link_check_status_handler', course_key)
+
+    # an _accept URL parameter will be preferred over HTTP_ACCEPT in the header.
+    requested_format = request.GET.get('_accept', request.META.get('HTTP_ACCEPT', 'text/html'))
+
+    if request.method == 'POST':
+        check_broken_links.delay(request.user.id, course_key_string, request.LANGUAGE_CODE)
+        return JsonResponse({'LinkCheckStatus': 1})
+    else:
+        # Only HTML request format is supported (no JSON).
+        return HttpResponse(status=406)
+
+
+@transaction.non_atomic_requests
+@require_GET
+@ensure_csrf_cookie
+@login_required
+@ensure_valid_course_key
+def link_check_status_handler(request, course_key_string):
+    """
+    Returns an integer corresponding to the status of a link check. These are:
```

Comment:
See comments below. I'm basically skeptical of the wisdom of trying to shoehorn all these disparate outcomes into a single int variable.